OptimoveSDK/Sources/Classes/Optimobile/AnalyticsHelper.swift

```diff
@@ -18,12 +18,14 @@ final class AnalyticsHelper {
     private var analyticsContext: NSManagedObjectContext?
     private var migrationAnalyticsContext: NSManagedObjectContext?
     private var finishedInitializationToken: NSObjectProtocol?
+    private let serialQueue : DispatchQueue
 
     // MARK: Initialization
 
     init(httpClient: KSHttpClient) {
         analyticsContext = nil
         migrationAnalyticsContext = nil
+        serialQueue = DispatchQueue(label: "com.optimove.optimobile.serial-q", target: .global(qos: .utility))
```

Comment:
For my own understanding - what's the reason for setting the target of `serialQueue` to `.global(qos: .utility)`? Do we need to dispatch our serial queue onto the global utility queue?

Reply:
utility qos means low priority. Do you think we need some other qos?

Reply:
Gotcha, `.utility` keeps it low priority. I'm not exactly sure of what the benefits would be, I can't offer a lot of experience here. I'd say I trust your solution - you have a better understanding of the problem and bigger picture